ecosystem/submarine-sends.md

```diff
@@ -0,0 +1,167 @@
+# Purpose
+
+Superchain interop enables new low-latency DeFi actions across chains. The protocol ensures integrity of the messages passed between chains, but additional app-level measures must also be in place to ensure users get the same single-chain DeFi experience.
+
+## Goals
+
+- Cross chain DeFi actions do not expose the user to additional 3rdparty risk
+
+## Non Goals
+
+- Masquerade submarine sends from regular transfer of funds.
+
+# Problem Statement + Context
+
+When using the `L2ToL2CrossDomainMessenger`, the target address and calldata is immediately made public when the message is sent. The outbox on the source chain becomes a visible mempool of transactions for a given destination.
+
+For many use cases, this can be fine but in DeFi can undermine fairness, due to front running. [Submarine Sends](https://libsubmarine.org/) is a way to mitigate these kinds of problems, where the transaction is made private until executed in a commit / reveal structure.
+
+# Proposed Solution
+
+We can take inspiration by replicating a commit / reveal scheme for data sent between chains. Enabling cross chain transactions between contracts where the intent remains hidden until the reveal on the destination chain.
+
+Unlike [Submarine Sends](https://libsubmarine.org/) which also provides the ability to make the commit transaction indistinguishable from a normal value transfer, the funds and commitment have to be accounted for and propogated through the `L2ToL2CrossDomainMessenger` which can allow for an external entinty to differentiate between a hashed cross domain submarine send and not. However this is not a problem since DeFi attacks rely on understanding the intent of the transaction which we show remains masked.
+
+- Maybe there's some additional design where the destination escrow address is fresh on every call, achieving the same property but that is left out of scope for now.
+
+## Commitment
+
+The commitment for a submarine send is the sensitive information in the outbound `SentMessage` event, `target` and `message`. For submarine sends where the same target & message can be used more than once, a salt can be provided to ensure unique commitments per transaction.
+
+```solidity
+bytes memory payload = abi.encodePacked(salt, target, message)
+bytes32 commitment = keccak256(payload)
+```
+
+By posting just the commitment, **only the the submitting user with knowledge of the preimage would be able to reveal the transaction** when relaying. If this user encounters issues between submission and relay, the preimage for that data is also lost.
+
+To solve for this, the payload is encrypted and posted as calldata to the submitting transaction. By using ECDH, Elliptic Curve Diffie-Hellman, the data can be encrypted against any public key, allowing for the user to designate a 3rdparty to conduct the relay on their behalf.
```

Comment:
I'd suggest leaving this preimage posting unspecified. If a user wants to post it encrypted onchain they can, but many implementations may prefer an out of band dissemination

Reply:
yea that's a good point! This all can be encoded as opaque bytes represented with 1 argument. Everything else should be good